### PR TITLE
DM-5922: Rework camera geometry to use the replacement for XYTransform

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: python/lsst/meas/algorithms/gaussianPsfFactory.py [hunk 1]
```diff
@@ -53,8 +53,8 @@ class GaussianPsfFactory(Config):
         check=isPositive,
     )
     sizeFactor = Field(
-        doc="Kernel size as a factor of fwhm (dimensionless); "
-        + "size = sizeFactor * fwhm; ignored if size is not None",
+        doc = "Kernel size as a factor of fwhm (dimensionless); " +
```
Comment (timj) [inline python/lsst/meas/algorithms/gaussianPsfFactory.py]: I didn't think the spaces should be there for keyword args?

Comment (timj) [inline python/lsst/meas/algorithms/gaussianPsfFactory.py]: Also, I don't think the space needs to be there at all does it?

Comment (r-owen) [inline python/lsst/meas/algorithms/gaussianPsfFactory.py]: I believe spaces around = for keyword arguments are:
- encouraged for keyword arguments that are on their own line
- discouraged for other cases

Comment (TallJimbo) [reply python/lsst/meas/algorithms/gaussianPsfFactory.py]: I checked, and @r-owen is correct according to the style guide.  This makes me sad; I don't think whitespace around `=` should be affected at all by how we split lines.  But I also don't have the energy to RFC a change.

File: python/lsst/meas/algorithms/objectSizeStarSelector.py [hunk 2]
```diff
@@ -368,10 +368,9 @@ def selectStars(self, exposure, sourceCat, matches=None):
         dumpData = lsstDebug.Info(__name__).dumpData                   # dump data to pickle file?
 
         detector = exposure.getDetector()
-        pixToTanXYTransform = None
+        pixToTanPix = None
         if detector is not None:
-            tanSys = detector.makeCameraSys(TAN_PIXELS)
-            pixToTanXYTransform = detector.getTransformMap().get(tanSys)
+            pixToTanPix = detector.getTransform(TAN_PIXELS)
```
Comment (TallJimbo) [inline python/lsst/meas/algorithms/objectSizeStarSelector.py]: Now that I'm seeing this interface in action in real code, I'm a bit bothered by how difficult it is to see what's going on.  This case is saved by the helpful `pixToTanPix` variable name, but I think `getTransform` should be replaced by a function that really says what's going on:
 - what other coordinate system is involved ("PIXELS")?
 - are we going from the given cordinate system ("TAN_PIXELS") to the implicit one ("PIXELS"), or the reverse?

To keep the meaning of the method the same, I'd recommend calling it something like `getTransformFromPixelsTo(target)`.  It's a lot more verbose, but a lot more readable.

Of course, this is all pre-existing, but it seems like a good time to fix it.

Comment (r-owen) [reply python/lsst/meas/algorithms/objectSizeStarSelector.py]: I agree. Depending on how widely used this is, I'll either go back to `detector.getTransformMap().getTransform(fromSys, toSys)` or add a convenience method `detector.getTransform`

**Comment by r-owen** [comment]
I fixed the primary complaint (which was well justified) and am much happier with the code now.

**Pull request merged**